plugins/base/src/main/kotlin/DokkaBase.kt

```diff
@@ -55,25 +44,16 @@ class DokkaBase : DokkaPlugin() {
     val externalLocationProviderFactory by extensionPoint<ExternalLocationProviderFactory>()
     val outputWriter by extensionPoint<OutputWriter>()
     val htmlPreprocessors by extensionPoint<PageTransformer>()
-    val kotlinAnalysis by extensionPoint<KotlinAnalysis>()
```

Comment:
Makes a lot of sense to think about this, yes! I've thought about it for some time and looked at the code, here's my proposal:

For now, we merge it as is, so that I don't block other work that depends on this PR. However, before 1.9.0, we must update the documentation - while doing that, we'll create a separate wiki page or an issue with the migration path, explaining the whole situation and what to do about it.

Once the issue/wiki with the explanation exists, we can add some of those types (like `KotlinAnalysis`, `ExternalClasslikesTranslator`) back, so that the code more or less compiles , and deprecate them with warning, giving the link to the issue/wiki page in the deprecation message.

Hopefully, people who are using it will come to the issue and report their use cases that we should provide stable API for